pkgs/top-level/all-packages.nix

```diff
@@ -7350,27 +7280,37 @@ with pkgs;
 
   glibc = callPackage ../development/libraries/glibc {
     installLocales = config.glibc.locales or false;
-    gccCross = null;
+    gccCross =
+      if crossSystem != null
+      then buildPackages.gccCrossStageStatic
+      else null;
+    inherit (buildPackages) linuxHeaders;
   };
 
   glibc_memusage = callPackage ../development/libraries/glibc {
     installLocales = false;
     withGd = true;
   };
 
-  glibcCross = forcedNativePackages.glibc.override {
-    gccCross = gccCrossStageStatic;
-    linuxHeaders = linuxHeadersCross;
-  };
-
-  # We can choose:
-  libcCrossChooser = name: if name == "glibc" then glibcCross
-    else if name == "uclibc" then uclibcCross
-    else if name == "msvcrt" then windows.mingw_w64
-    else if name == "libSystem" then darwin.xcode
-    else throw "Unknown libc";
+  libc =
+    if crossSystem != null
```

Comment:
We shouldn't need crossSystem here. Why not just use stdenv.hostPlatform.libc? libc should come from pkgsTargetTarget right?